src/org/zaproxy/zap/extension/websocket/pscan/scripts/ScriptsWebSocketPassiveScanner.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.websocket.pscan.scripts;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptType;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+import org.zaproxy.zap.extension.websocket.ExtensionWebSocket;
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketAlertWrapper;
+import org.zaproxy.zap.extension.websocket.alerts.WebSocketPassiveScanAlert;
+import org.zaproxy.zap.extension.websocket.pscan.WebSocketPassiveScanThread;
+import org.zaproxy.zap.extension.websocket.pscan.WebSocketPassiveScannerManager;
+import org.zaproxy.zap.extension.websocket.pscan.WebSocketPluginPassiveScanner;
+
+import java.lang.reflect.UndeclaredThrowableException;
+import java.util.List;
+
+/**
+ * Implement a plugin for Passive Scanning Scripting.
+ */
+public class ScriptsWebSocketPassiveScanner extends WebSocketPluginPassiveScanner {
+    
+    private static final Logger LOGGER = Logger.getLogger(ScriptsWebSocketPassiveScanner.class);
+    public static final String PLUGIN_NAME = "WS.ScriptPassiveScan";
+    public static final int PLUGIN_ID = 25379412;
```

Comment:
Plugin ids should be registered in https://github.com/zaproxy/zaproxy/blob/develop/docs/scanners.md
Maybe start from 110000 - ideally each rule should have a unique one